### PR TITLE
[wip][train] clean up AIR references

### PR DESCRIPTION
**Pull request by matthewdeng**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-overview/installation.rst [hunk 1]
```diff
@@ -370,10 +370,11 @@ To install Ray libraries, use ``pip`` as above or ``conda``/``mamba``.
 
 .. code-block:: bash
 
-  conda install -c conda-forge "ray-air"    # installs Ray + dependencies for Ray AI Runtime
+  conda install -c conda-forge "ray-data"   # installs Ray + dependencies for Ray Data
```
Comment (krfricke) [inline doc/source/ray-overview/installation.rst:373]: Should we remove `ray-air` from conda? https://github.com/conda-forge/ray-packages-feedstock/blob/main/recipe/meta.yaml#L109

File: python/ray/train/examples/mlflow_simple_example.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 from ray import train
 from ray.train import ScalingConfig, RunConfig
 from ray.train.torch import TorchTrainer
-from ray.air.integrations.mlflow import MLflowLoggerCallback
+from ray.tune.logger.mlflow import MLflowLoggerCallback
```
Comment (krfricke) [inline python/ray/train/examples/mlflow_simple_example.py:5]: More for a follow-up, but it's pretty annoying that we have to import from tune here

**Review by krfricke — APPROVED** [review]
LGTM

File: python/ray/train/examples/mlflow_simple_example.py (hunk 2 above)
Comment (matthewdeng) [reply python/ray/train/examples/mlflow_simple_example.py:5]: Yeah agreed, though this does subclass `ray.tune.logger.LoggerCallback` so I think it's okay for now.

**Review by c21 — APPROVED** [review]

**Pull request merged**